Add eclipse docs (#365)
zaproxy/zaproxy-website: site/content/docs/developer/building-zap-with-eclipse.md

---
title: "Building ZAP with Eclipse"
description: "This guide will get you started with building and running ZAP in the Eclipse IDE."
tags: 
- guide
- tutorial
type: page
date: "2021-03-18"
---

> If you've want to make changes to ZAP using the Eclipse IDE then you are in the right place.

Reviewer: ```suggestion
If you want to make changes to ZAP using the Eclipse IDE then you are in the right place.
```

## Preparation

> You will need to have followed the [Quick Start Guide to Building ZAP](../quick-start-build/) and installed [Eclipse](https://www.eclipse.org/).

Reviewer: ```suggestion
You will need to have followed the [Quick Start Guide to Building ZAP](../quick-start-build/) and installed [Eclipse](https://www.eclipse.org/). We would suggest the "Eclipse IDE for Java Developers" [package](https://www.eclipse.org/downloads/packages/).
```

## Import the ZAP Repos
In Eclipse select the 'File' menu, then 'Import...', and then 'Gradle' &#8594; 'Existing Gradle Project':

> ![Eclipse Gradle Import](/img/docs/developer/eclipse-import.png)

Reviewer: I'm trusting these paths are correct, they don't render on GitHub.

In the following dialog enter the full directory name for one of the ZAP repos you cloned before and click on 'Finish'.

Repeat this process until you have imported all of the ZAP repos.

## Running ZAP
In the Eclipse Project Explorer navigate to:
* zaproxy / zap / src / main / java / org / zaproxy / zap

Right click `ZAP.java` and select 'Run As' &#8594; 'Java Application'.

ZAP will now start but it will not be set up quite right and may show warning dialogs, so close it and go back to Eclipse.

Select the 'Run' menu, then 'Run Configurations...' and then the `ZAP` configuration.

Select the 'Arguments' tab, change the 'Working directory' from 'Default' to 'Other' and enter: `${workspace_loc:zaproxy/zap/src/main/dist/}`

> ![Eclipse Gradle Import](/img/docs/developer/eclipse-configs.png)

Reviewer: ```suggestion
![Eclipse Run Configurations](/img/docs/developer/eclipse-configs.png)
```

Run this configuration and ZAP should now start with all of the add-ons correctly configured.

## Running Gradle Tasks
You can run the ZAP Gradle tasks from the command line or from within Eclipse, the end result will be the same.

Eclipse has a 'Gradle Tasks' tab that lists all of the tasks available to you ordered by project. 
Note that the add-ons in zap-extensions all have their own set of tasks in addition to the top level ones.

## Tips and Tricks

### Refresh Gradle Project

> Sometimes Eclipse will unexpected fail to compile ZAP code. This can occur if you switch between git branches which have different dependencies.

Reviewer: ```suggestion
Sometimes Eclipse will unexpectedly fail to compile ZAP code. This can occur if you switch between git branches which have different dependencies.
```

If this happens to you then right click the relevant project in the Eclipse Project Explorer and select 'Gradle' &#8594; 'Refresh Gradle Project'

### Package Presentation

The default way in which Eclipse displays the package hierarchy does not work very well for ZAP, at least in our opinion.

To change this click on the buttton with 3 vertical dots at the top of the Package Explorer, select 'Package Presentation' &#8594; 'Hierarchical'.

> ![Eclipse Gradle Import](/img/docs/developer/eclipse-package.png)

Reviewer: ```suggestion
![Eclipse Package Presentation](/img/docs/developer/eclipse-package.png)
```

tags: 
- guide
- tutorial
type: page
date: "2021-03-18"
---

If you want to make changes to ZAP using the Eclipse IDE then you are in the right place.

## Preparation
You will need to have followed the [Quick Start Guide to Building ZAP](../quick-start-build/) and installed a version of [Eclipse](https://www.eclipse.org/) suitable for Java development. We would suggest the "Eclipse IDE for Java Developers" [package](https://www.eclipse.org/downloads/packages/).

## Import the ZAP Repos
In Eclipse select the 'File' menu, then 'Import...', and then 'Gradle' &#8594; 'Existing Gradle Project':

![Eclipse Gradle Import](/img/docs/developer/eclipse-import.png)

In the following dialog enter the full directory name for one of the ZAP repos you cloned before and click on 'Finish'.

Repeat this process until you have imported all of the ZAP repos.

It is worth noting that some of the zap-extension add-ons may fail to compile if you do not have the required environment for them. This is to be expected and is not a problem unless you want to work on those add-ons, in which case see their READMEs for more details.

## Running ZAP

> In the Eclipse Project Explorer navigate to:

Reviewer 2: Alternatively (or additionally) we could suggest using the `zaproxy:run` gradle task?

Reviewer 3: That's counter-productive IMO, if we are using the IDE better make full use of it.

Reviewer 2: +1 but we can also use gradle tasks from within the IDE?

Author: Ha! I've added a note saying you can do that. Whats the verdict - include or remove?

Reviewer 3: Yes, that's necessary when installing or deploying the add-ons but running ZAP shouldn't be needed since the IDE allows to run it (and configure it more easily, plus debug).

ZAP will now start but it will not be set up quite right and may show warning dialogs, so close it and go back to Eclipse.

Select the 'Run' menu, then 'Run Configurations...' and then the `ZAP` configuration.

Select the 'Arguments' tab, change the 'Working directory' from 'Default' to 'Other' and enter: `${workspace_loc:zaproxy/zap/src/main/dist/}`

![Eclipse Run Configuration](/img/docs/developer/eclipse-config-run.png)

Run this configuration and ZAP should now start with all of the add-ons correctly configured.

## Running Gradle Tasks
You can run the ZAP Gradle tasks from the command line or from within Eclipse, the end result will be the same.

Eclipse has a 'Gradle Tasks' tab that lists all of the tasks available to you ordered by project. 
Note that the add-ons in zap-extensions all have their own set of tasks in addition to the top level ones.

## Debugging Add-ons
You can debug code in the zaproxy repo without making any changes, but you will need to tweak your configuration in order to debug add-ons.

Select the ZAP runtime configuration as before, switch to the 'Source' tab and click on the 'Add...' button.

> Select 'Java Project' and click 'OK' then select the add-ons which you want to be able to debug before clicking on the 'OK' button.

Reviewer 2: ```suggestion
Select 'Java Project', click on 'OK', and then select the add-ons which you want to be able to debug before proceeding.
```
